vm/src/format.rs

```diff
@@ -245,14 +247,17 @@ fn parse_format_spec(text: &str) -> FormatSpec {
     let (width, after_width) = parse_number(after_zero);
     let (grouping_option, after_grouping_option) = parse_grouping_option(after_width);
     let (precision, after_precision) = parse_precision(after_grouping_option);
-    let (format_type, _) = parse_format_type(after_precision);
+    let (format_type, after_format_type) = parse_format_type(after_precision);
+    if !after_format_type.is_empty() {
+        return Err("Invalid format spec");
+    }
```

Comment:
Maybe add some tests that ensure that this properly fails with characters left after the format type.

Reply:
:+1: